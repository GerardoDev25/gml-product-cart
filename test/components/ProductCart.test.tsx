import React from 'react';
import rendered from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product1 } from './data/product';

const { act } = rendered;

describe('test in ProductCart', () => {
  test('should to show the component correctly with the img costumed', () => {
    const wrapper = rendered.create(
      <ProductCard product={product1}>{() => <h1>ProductCart</h1>}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should to increase the counter', () => {
    const wrapper = rendered.create(
      <ProductCard product={product1}>
        {({ count, increaseBy }) => (
          <>
            <h1>Prodcut Cart</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(1)}></button>
          </>
        )}
      </ProductCard>
    );
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();

    act(() => {
      (tree as any).children[2].props.onClick();
    });

    tree = wrapper.toJSON();

    expect((tree as any).children[1].children[0]).toBe('0');
  });
});
