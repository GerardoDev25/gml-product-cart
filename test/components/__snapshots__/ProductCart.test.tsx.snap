// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in ProductCart should to increase the counter 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Prodcut Cart
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`test in ProductCart should to show the component correctly with the img costumed 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCart
  </h1>
</div>
`;
